lib/fedex/request/tracking_information.rb

```diff
@@ -36,12 +36,23 @@ def process_request
             Fedex::TrackingInformation.new(details)
           end
         else
-          error_message = if response[:track_reply]
-            response[:track_reply][:notifications][:message]
+          if response[:track_reply]
+            error_message = response[:track_reply][:notifications][:message]
+            # These error codes come from the Fedex Webservices Developer Guide (2019)
+            # (https://www.fedex.com/us/developer/downloads/pdf/2019/FedEx_WebServices_DevelopersGuide_v2019.pdf
+            #  via https://www.fedex.com/en-us/developer/web-services/process.html#documentation).
+            error_code = response[:track_reply][:notifications][:code]
+            if invalid_tracking_number? error_code
+              raise InvalidTrackingNumberError, "[#{error_code}] #{error_message} (#{@package_id})"
```

Comment:
after millennia of suffering and countless lost alerts we have arrived...transparent errors, what a time to be alive!